microconfig-cli/src/main/java/io/microconfig/MicroconfigMain.java

```diff
@@ -39,42 +47,59 @@ public static void main(String... args) {
         }
 
         File rootDir = params.rootDir();
-        File destinationDir = params.destinationDir();
-        String env = params.env();
+        String destinationDir = params.destinationDir();
+        Set<String> environments = params.environments();
         List<String> groups = params.groups();
         List<String> services = params.services();
         boolean stacktrace = params.stacktrace();
         boolean jsonOutput = params.jsonOutput();
 
-        new MicroconfigMain(rootDir, destinationDir, env, groups, services, stacktrace, jsonOutput).build();
+        new MicroconfigMain(rootDir, destinationDir, environments, groups, services, stacktrace, jsonOutput).build();
     }
 
     private void build() {
         try {
-            long startTime = currentTimeMillis();
+            enableLogger(!jsonOutput);
             doBuild();
-            announce("\nGenerated [" + env + "] configs in " + (currentTimeMillis() - startTime) + "ms");
         } catch (RuntimeException e) {
             if (stacktrace || e.getMessage() == null) {
                 throw e;
             }
-
             error(e.getMessage());
             exit(-1);
         }
     }
 
     private void doBuild() {
-        enableLogger(!jsonOutput);
+        Set<String> envsToBuild = chooseEnvironments();
 
-        MicroconfigRunner runner = new MicroconfigRunner(rootDir, destinationDir);
+        envsToBuild.forEach(e -> {
+            long startTime = currentTimeMillis();
+            String filePath = envsToBuild.size()>1 ? String.format("%s/%s", destinationDir, e) : destinationDir;
```

Comment:
envsToBuild.size()>1 is wrong. cause you can run -envs prod or - envs * and you have only 1 env.